src/functions.js

```diff
@@ -36,40 +62,75 @@ function cities( capitals, formatter ){
   // such as 'Paris is the capital of France'
 
   // apply formatter to each object in capitals array and return an array of resulting sentences
+
+var mappedArray = capitals.map(function formatter(myArrItems) {
+      return (myArrItems.city + " is the capital of " + myArrItems.country)});
+      return  mappedArray;
 }
 
 function largerThanTen( numbers ){
   // numbers is an array of numbers
   // return a new array that contains only numbers from the input array which are greater than 10
   // hint: filter
+
+  function filterItems(myArrItems){
+    return  myArrItems > 10;
+ }
+
+    return filtered = numbers.filter(filterItems); 
+
 }
 
 function even( numbers ){
   // numbers is an array of numbers
   // return a new array that contains only even numbers from the input array
   // hint: you may want to use the modulus operator '%'
+
+  function filterEven(myArrItems){
+    return  myArrItems % 2 === 0;
+ }
+
+    return even = numbers.filter(filterEven); 
+
+
 }
 
 function findTheNeedle( words ){
   // words is an array of words
   // return the index of the word 'needle'
   // hint: indexOf
+
+  return words.indexOf('needle');
 }
 
 function findLargest( numbers ){
   // numbers is an array of numbers
   // return the largest number from that array
+  
+  return Math.max.apply(null, numbers);
 }
 
 function addAllnumbers( numbers ) {
   // numbers is an array of numbers
   // return the sum of all the numbers in the array
+  var result = numbers.reduce(function(accumulator, currentValue) {
+    return accumulator + currentValue;
+});
+
+    return result;
+ 
 }
 
 function average( things ) {
   // things is an array of numbers and strings
   // return the average of all the numbers
   // be sure to exclude the strings
+  var sum;
+  things.forEach(function(n){
+    sum += n ; 
+    avg = sum / things.length;
```

Comment:
there is no need to calculate the average at each iteration, you can move this outside of the `forEach`. Furthermore you don't need the length of the whole array `things`, but just the subarray composed by numbers😉